### ISSUE TITLE
Dark mode for mitmweb

### ISSUE THREAD

**Issue by oxr463** [mitmproxy/mitmproxy#3886]
Love the web ui, but the all white / really bright interface is not the best on the eyes in the evenings.

See also: https://postwoman.io

**Comment by mhils** [comment]
Thanks for the suggestion! I'm happy to merge a PR that adds dark-mode CSS depending on system settings, but it's nothing I see one of the core team spend time on in at the moment.

**Comment by Prinzhorn** [comment]
Just my 2 cents for anyone that wants to tackle this: I've made good experiences using CSS variables and `@media (prefers-color-scheme: dark)`. Optionally you can force an overwrite of the media query using a class added via JavaScript (and maybe persisted in `localStorage` for extra swag).

```css
:root {
  --text: #000;
  --background-default: #fff;
}

@media (prefers-color-scheme: dark) {
  :root {
    --text: #fff;
    --background-default: #222;
  }
}

body {
  color: var(--text);
  background: var(--background-default);
}
```

**Comment by oxr463** [comment]
Trying to figure out how to toggle the theme. I'm missing something on the server-side, (See: https://github.com/mitmproxy/mitmproxy/issues/3925)

**Comment by Prinzhorn** [comment]
> Trying to figure out how to toggle the theme. I'm missing something on the server-side, (See: #3925)

I personally don't think the server needs to be involved at all. Toggling is something that happens in the browser. If persistence is needed it can be done via `localStorage`. Or is there a use-case that requires server interaction that I'm missing?

**Comment by oxr463** [comment]
How can I execute a function based on the `SETTINGS_UPDATE` action?

![image](https://user-images.githubusercontent.com/43783393/82447156-df835a00-9a75-11ea-9a28-f684aaea9838.png)

The `localStorage` part is fine, I can easily read/write values there. But I don't know where in [`js`](https://github.com/mitmproxy/mitmproxy/tree/master/web/src/js) to do that.

**Comment by AClon314** [comment]
Share the css style, you can use **stylus** browser extension to apply that on http://127.0.0.1
```css
@-moz-document url-prefix("http://127.0.0.1") {
    @media (prefers-color-scheme: dark) {
        *:not(ul) {
            background: none !important;
        }
        :root {
            --text: #fff;
            --background-default: #000;
        }
        body,button,pre,a {
            color: var(--text) !important;
        }
        pre,
        body,
        header,
        .modal-content {
            background: var(--background-default) !important;
        }
        .selected {
            background-color-color: #333 !important;
        }
    }
}
```

**Comment by lups2000** [comment]
We have a dark mode now, thanks to @sleeyax 🚀 🥳

**Comment by KaKi87** [comment]
It took AI for this 2020 request to be addressed haha 🤣

**Comment by lups2000** [comment]
> It took AI for this 2020 request to be addressed haha 🤣

exactly 😅

**Comment by sleeyax** [comment]
> It took AI for this 2020 request to be addressed 

AI definitely helped with the implementation. Last year I started on a complete rewrite with a new component and a Tailwind-based theming system, all handwritten. It ended up being too large of a change to land, though. 

For this version I designed the theme by hand, then used AI to find the remaining hard-coded styles scattered throughout the codebase. It made the tedious parts much faster and finally got the feature over the finish line. 🙂
